### PR TITLE
feat(2267): refactor collection page and apply lazy loading

### PR DESCRIPTION
**Pull request by klu909**
## Context

Collection page does not render until metrics for all pipeline are loaded.
## Objective

Move metric query to each individual pipeline card/row component so collection page render without waiting for metrics 
## References

https://github.com/screwdriver-cd/screwdriver/issues/2267

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/collection-table-row/template.hbs [hunk 1]
```diff
@@ -8,15 +8,21 @@
...
+  <span title={{branch}}>{{branch}}</span>
 </td>
 <td class="status">
   {{#link-to "pipeline" pipeline.id title=pipeline.scmRepo.name}}
     {{fa-icon lastEventInfo.icon class=lastEventInfo.statusColor}}
   {{/link-to}}
-  <a href={{lastEventInfo.commitUrl}}>
-    {{lastEventInfo.sha}}
-  </a>
+  {{#if queryError}}
```
Comment (adong) [inline app/components/collection-table-row/template.hbs:17]: What `queryError` could be?

File: app/components/pipeline-card/component.js [hunk 2]
```diff
@@ -11,20 +15,46 @@ export default Component.extend({
   pipelineSelected: false,
   classNames: ['pipeline-card'],
   reset: false,
-
+  queryError: false,
+  updated: and('eventsInfo', 'lastEventInfo'),
```
Comment (adong) [inline app/components/pipeline-card/component.js:19]: Can we rename `updated` to `hasBothEventsAndLatestInfo` ? Latter is more intuitive.

**Review by adong — CHANGES_REQUESTED** [review]
Please add some descriptions to this PR.

File: app/components/collection-table-row/template.hbs (hunk 1 above)
Comment (klu909) [reply app/components/collection-table-row/template.hbs:17]: if store query fails, for example getting 402 or 404 for whatever reason

Comment (klu909) [reply app/components/collection-table-row/template.hbs:17]: it's just a boolean

**Review by adong — APPROVED** [review]

**Pull request merged**